app/models/miq_worker/systemd_common.rb

```diff
@@ -0,0 +1,144 @@
+        File.write(service_config_file_path, service_settings_file) unless service_config_file_path.exist?
+      end
+
+      def service_base_name
+        "#{minimal_class_name.underscore.tr("/", "_")}"
+      end
+
+      def service_name
+        singleton_worker? ? "#{service_base_name}" : "#{service_base_name}@"
+      end
+
+      def service_file_name
+        "#{service_name}.service"
+      end
+
+      def slice_name
+        "cfme-#{service_base_name}.slice"
+      end
+
+      def service_file_path
+        systemd_unit_dir.join(service_file_name)
+      end
+
+      def service_config_name
+        "#{service_name}.d"
+      end
+
+      def service_config_path
+        systemd_unit_dir.join(service_config_name)
+      end
+
+      def service_config_file_path
+        service_config_path.join("settings.conf")
+      end
+
+      def target_file_name
+        "#{service_base_name}.target"
+      end
+
+      def target_file_path
+        systemd_unit_dir.join(target_file_name)
+      end
+
+      def systemd_unit_dir
+        Pathname.new("/etc/systemd/system")
+      end
+
+      def target_file
+        <<~TARGET_FILE
+          [Unit]
+          PartOf=cfme.target
+        TARGET_FILE
+      end
+
+      def unit_file
+        <<~UNIT_FILE
+          [Unit]
+          PartOf=#{target_file_name}
+          [Install]
+          WantedBy=#{target_file_name}
+          [Service]
+          Environment=HOME=/root
+          WorkingDirectory=#{working_directory}
+          ExecStart=/bin/bash -lc '#{exec_start}'
+          Restart=always
+          Slice=#{slice_name}
+        UNIT_FILE
+      end
+
+      def working_directory
+        # TODO pull this dynamically
+        "/var/www/miq/vmdb"
+      end
+
+      def exec_start
+        "exec ruby lib/workers/bin/run_single_worker.rb #{self.name} #{run_single_worker_args}"
+      end
+
+      def run_single_worker_args
+        "--heartbeat"
+      end
+
+      def service_environment_variables
+        # Override this in a child class to add env vars
+        ["GUID=#{guid}"]
+      end
+
+      def service_settings_file
+        <<~WORKER_SETTINGS_FILE
+          [Service]
+          #{service_environment_variables.map { |env_var| "Environment=#{env_var}" }.join("\n")}
+          MemoryHigh=#{worker_settings[:memory_threshold].bytes}
+          TimeoutStartSec=#{worker_settings[:starting_timeout]}
+          TimeoutStopSec=#{worker_settings[:stopping_timeout]}
+        WORKER_SETTINGS_FILE
+      end
+    end
+
+    def enable_systemd_unit(runtime: false, replace: true)
+      systemd.EnableUnitFiles([unit_name], runtime, replace)
+    end
+
+    def disable_systemd_unit(runtime: false)
+      systemd.DisableUnitFiles([unit_name], runtime)
+    end
+
+    def start_systemd_unit(mode: "replace")
+      systemd.StartUnit(unit_name, mode)
+    end
+
+    def stop_systemd_unit(mode: "replace")
+      systemd.StopUnit(unit_name, mode)
+    end
```

Comment:
We should probably replace https://github.com/ManageIQ/linux_admin/blob/master/lib/linux_admin/service/systemd_service.rb with this also. Not sure if you want to take that on for this though ...

Reply:
Yeah that's absolutely something I intend to do

Reply:
Should we use LinuxAdmin here?

Reply:
So you want to replace AwesomeSpawn with dbus-systemd in LinuxAdmin first?